src/tools/quemap/lightmap.c

```diff
@@ -620,6 +620,87 @@ static _Bool NudgeSamplePosition(const vec3_t in, const vec3_t normal, const vec
 	return Light_PointPVS(out, pvs);
 }
 
+/**
+ * @brief Trace a ray and gather indirect light from surfaces
+ */
+static void GatherSampleBounceTrace(int8_t bounce, const vec3_t center, const vec3_t pos, const vec3_t normal, const vec_t dist, vec_t *sample) {
+
+	vec3_t end;
+	vec_t dot, sqdist;
+	vec3_t color;
+	vec_t hardness;
+	vec3_t bounce_dir, bounce_pos;
+	vec_t bounce_dist;
+	cm_trace_t trace;
+
+	// get a trace
+
+	VectorScale(normal, dist, end);
+
+	Light_Trace(&trace, pos, end, CONTENTS_SOLID);
+
+	if (!trace.surface) {
+		return;
+	}
+
+	if (trace.start_solid) {
+		return;
+	}
+
+	// distance squared and trace hit normal dot product
+
+	sqdist = trace.fraction * trace.fraction;
```

Comment:
This isn't the distance squared, it's the trace fraction squared. Subtract the trace start and end points, and take `VectorLengthSquared` of that if you want distance squared.